### DISCUSSION TITLE
How to extract vendors into separate bundle with Parcel manualSharedBundles

### DISCUSSION THREAD

**Discussion by eugene-khyst** [parcel-bundler/parcel#9620] [Q&A]
In Parcel 2.12.0 a [Manual shared bundles](https://parceljs.org/blog/v2-12-0/#manual-shared-bundles) feature was introduced

Can Manual shared bundles be used to extract vendors (React, Ant Design, etc.) to a separate bundle?
Like it can be done with WebPack and [SplitChunksPlugin](https://webpack.js.org/plugins/split-chunks-plugin/)?

The following config doesn't work, there is still a single huge bundle:

```json
"@parcel/bundler-default": {
    "manualSharedBundles": [
      {
        "name": "vendor",
        "root": "src/root.tsx",
        "assets": [
          "node_modules/**/*"
        ],
        "types": [
          "js"
        ]
      }
    ]
  },
```

Or is creating a dummy `duplicate.tsx` still the only way to extract vendors to a separate bundle as described in this [discussion](https://github.com/parcel-bundler/parcel/discussions/7733)?

**Comment by devongovett** [comment]
In your configuration, you have set the `root` option, so this will only apply to `node_modules` that are imported directly from `src/root.tsx`. So if you wanted this vendor bundle to apply across all dependencies in your whole app, you'd need to remove `root` I think.

**Reply by eugene-khyst** [reply]
I tried to remove `"root": "src/root.tsx"`:
```
"@parcel/bundler-default": {
    "manualSharedBundles": [
      {
        "name": "vendor",
        "assets": [
          "node_modules/**/*"
        ],
        "types": [
          "js"
        ]
      }
    ]
  },
```
Still doesn't give the desired result.

**Comment by shepmaster** [comment]
I was also trying and failing to do this. Here's my reproduction steps:

**package.json**

```json
{
  "name": "repro",
  "version": "1.0.0",
  "@parcel/bundler-default": {
    "manualSharedBundles": [
      {
        "name": "vendor",
        "assets": [
          "node_modules/**/*"
        ],
        "types": [
          "js"
        ]
      }
    ]
  },
  "source": "./src/index.js",
  "targets": {
    "main": {
      "includeNodeModules": true
    }
  },
  "dependencies": {
    "add-two-numbers-for-testing": "^1.0.0"
  },
  "devDependencies": {
    "parcel": "^2.12.0"
  }
}
```

**src/index.js**

```js
import addNumber from 'add-two-numbers-for-testing';

console.log(addNumber(1, 2));
```

---

I build via `pnpm parcel build` and then check in `dist/main/`. There's no bundle for `vendor` in there, and inspecting the contents of `dist/main/index.js` show that it contains the code from the "add-two-numbers-for-testing" dependency.
